src/hotspot/os_cpu/linux_aarch64/pauth_linux_aarch64.inline.hpp

```diff
@@ -46,26 +46,26 @@ inline address pauth_strip_pointer(address ptr) {
   return result;
 }
 
-// Sign a return value, using the given modifier.
+// Sign a return value, using value zero as the modifier.
 //
-inline address pauth_sign_return_address(address ret_addr, address sp) {
+inline address pauth_sign_return_address(address ret_addr) {
   if (VM_Version::use_rop_protection()) {
     // A pointer cannot be double signed.
     guarantee(pauth_ptr_is_raw(ret_addr), "Return address is already signed");
     register address r17 __asm("r17") = ret_addr;
-    register address r16 __asm("r16") = sp;
+    register address r16 __asm("r16") = 0;
     asm (PACIA1716 : "+r"(r17) : "r"(r16));
```

Comment:
Can we use PACIZA or PACIA here so we don't force the use of r16/r17?

Reply:
Thanks for pointing it out.

`pacia1716` was used in the initial implementation of pac-ret, mainly because we used `fp` as the pac modifier then and `pacia x30, fp` doesn't belong to the NOP instruction space.

Now we use "zero const" as the modifier and we'd better use `paciaz/autiaz` as it's consistent with our usages in macro-assembler helpers i.e. `protection_return_address()` and `authenticate_return_address()`.

Updated in the latest commit. Please help take another look at it. Thanks.